pages/snake/index.js

```diff
@@ -0,0 +1,372 @@
+import Player from "@/components/SnakeGame/Player/Player";
+import StyledLink from "@/components/StyledLink/StyledLink";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+
+const StyledSnakePage = styled.main`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  gap: 30px;
+  width: 100vw;
+  height: 100vh;
+  background: #f1f3c2;
+  @media screen and (min-width: 600px) {
+    gap: 5vh;
+  }
+  @media screen and (min-width: 900px) {
+    gap: 10vh;
+  }
+`;
+
+const StyledGameField = styled.div`
+  display: grid;
+  place-content: center;
+  width: 300px;
+  min-height: 300px;
+  background: #008000;
+  background-image: url("https://www.transparenttextures.com/patterns/checkered-light-emboss.png");
+  border: 2px solid #000000;
+  border-radius: 20px;
+  position: relative;
+  border: 5px solid #000000;
+  @media screen and (min-width: 600px) {
+    transform: scale(1.2);
+  }
+  @media screen and (min-width: 900px) {
+    transform: scale(1.5);
+  }
+`;
+
+const StyledScoreAndButtonContainer = styled.div`
+  position: relative;
+`;
+
+const StyledScoreContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  width: 300px;
+  height: 100px;
+  font-size: 1.2rem;
+  font-weight: 800;
+`;
+
+const StyledButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-top: 10vh;
+`;
+
+const StyledIndicatorContainer = styled.div`
+  position: absolute;
+  left: -175px;
+  top: 60px;
+  transform: rotate(270deg);
+  width: 300px;
+
+  @media screen and (min-width: 600px) {
+    left: -200px;
+  }
+`;
+
+const StyledGameOver = styled.h1`
+  position: absolute;
+  top: 30%;
+  left: 25%;
+`;
+
+const StyledHowToPlay = styled.div`
+  display: grid;
+  place-content: center;
+  gap: 20px;
+  background: var(--secondary-color);
+  font-size: 0.8rem;
+  line-height: 2;
+  width: 100%;
+  padding: 20px;
+  border-radius: 20px;
+  position: absolute;
+  top: -20px;
+  z-index: 2;
+  @media screen and (min-width: 900px) {
+    display: block;
+    width: 120px;
+    right: -140px;
+    top: 20px;
+    font-size: 0.6rem;
+    background: none;
+    padding: 0;
+    line-height: 1.5;
+  }
+  @media screen and (min-width: 1200px) {
+    width: 200px;
+    font-size: 0.8rem;
+    right: -230px;
+  }
+`;
+
+export default function SnakeGame({ currentPet, onUpdatePetIndicator }) {
+  const [gameOn, setGameOn] = useState(true);
+  const [playerPosition, setPlayerPosition] = useState({ x: 140, y: 140 });
+  const [children, setChildren] = useState([]);
+  const [foodPosition, setFoodPosition] = useState({});
+  const [directions, setDirections] = useState({
+    currentDirection: "",
+    prevDirection: "",
+  });
+  const [scores, setScores] = useState({ score: 0, highscore: 0 });
+  const [instructions, setInstructions] = useState(false);
+  useEffect(() => {
+    setFoodPosition(generateNewFoodPosition());
+  }, []);
```

Comment:
Does the state need to be set with the function call inside a `useEffect`?

Can you not initialize the state like this:

```js
const [foodPosition, setFoodPosition] = useState( {x: Math.floor(Math.random() * 14) * 20,
      y: Math.floor(Math.random() * 14) * 20} )
```

If you did so, you would get rid of the `useEffect`